Sources/Rules/EmptyExtension.swift

```diff
@@ -0,0 +1,30 @@
+//
+//  EmptyExtension.swift
+//  SwiftFormat
+//
+// Created by manny_lopez on 7/29/24.
+//  Copyright © 2024 Nick Lockwood. All rights reserved.
+//
+
+import Foundation
+
+public extension FormatRule {
+    /// Remove empty, non-conforming, extensions
+    static let emptyExtension = FormatRule(help: "Remove empty, non-conforming, extensions") { formatter in
+        var emptyExtensions = [Declaration]()
+        formatter.forEachRecursiveDeclaration { declaration in
+            guard case let .type(_, open, body, _, _) = declaration,
+                  declaration.keyword == "extension",
+                  body.isEmpty,
+                  !open.contains(where: { $0 == .delimiter(":") })
```

Comment:
You can avoid the `case let .type` by accessing properties like `declaration.body` and `declaration.openTokens`

```suggestion
            guard
                  let declarationBody = declaration.body,
                  declaration.keyword == "extension",
                  declarationBody.isEmpty,
                  !declaration.openTokens.contains(where: { $0 == .delimiter(":") })
```

Reply:
thank you! Added.